gthockey-iOS/gthockey-iOS/CustomUIComponents/PillButton.swift

```diff
@@ -57,4 +58,36 @@ class PillButton: UIButton {
         fatalError("init(coder:) has not been implemented")
     }
 
+    // MARK: Loading Animation
+
+    private var originalButtonText: String?
+
+    private lazy var activityIndicator: UIActivityIndicatorView = {
+        let activityIndicator = UIActivityIndicatorView()
+        activityIndicator.translatesAutoresizingMaskIntoConstraints = false
+        activityIndicator.color = .black
+        addSubview(activityIndicator)
+
+        NSLayoutConstraint.activate([
+            activityIndicator.centerYAnchor.constraint(equalTo: self.centerYAnchor),
+            activityIndicator.centerXAnchor.constraint(equalTo: self.centerXAnchor)
+        ])
+
+        return activityIndicator
+    }()
+
+    func loading(_ isLoading: Bool) {
```

Comment:
Instead of being a function, can you set this to be a variable of the PillButton class? I was thinking you could do it similar to the `isEnabled` or `isHighlighted` attrs at the top of the class.